sqlx-core/src/postgres/types/interval.rs

```diff
@@ -0,0 +1,129 @@
+use std::mem;
+
+use byteorder::{NetworkEndian, ReadBytesExt};
+
+use crate::decode::Decode;
+use crate::encode::Encode;
+use crate::postgres::protocol::TypeId;
+use crate::postgres::{PgData, PgRawBuffer, PgTypeInfo, PgValue, Postgres};
+use crate::types::Type;
+use crate::Error;
+
+/// PostgreSQL INTERVAL type binding
+#[derive(Debug, PartialEq)]
+pub struct PgInterval {
+    pub months: i32,
+    pub days: i32,
+    pub microseconds: i64,
+}
+
+impl PgInterval {
+    pub fn new(months: i32, days: i32, microseconds: i64) -> Self {
```

Comment:
If we are making the fields of `PgInterval` public, there isn't much point to have the `new` here.